bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java

```diff
@@ -29,12 +29,6 @@
     public static final long CACHE_EXPIRE = System.currentTimeMillis() - 24 * 3600 * 1000;
     private static final String WORKSPACES_DIR_SYSTEM_PROPERTY = "jenkins.model.Jenkins.workspacesDir";
 
-    /**
-     * This system property is set by the bootstrap script created by appassembler Maven plugin
-     * to point to a local Maven repository.
-     */
-    public final File appRepo = new File(System.getProperty("app.repo"));
-
```

Comment:
I believe in case of fat jar, the **app.repo**  property will not be provided since repo is part of the jar itself. We've made it on-demand initialization with the help of a function below which will be executed when collecting jars from repo is required.